server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java

```diff
@@ -357,11 +357,16 @@ private FilterBuilder viewFilter(Collection<String> viewUuids) {
         .lookupIndex(ViewIndexDefinition.INDEX)
         .lookupType(ViewIndexDefinition.TYPE_VIEW)
         .lookupId(viewUuid)
-        .lookupPath(ViewIndexDefinition.FIELD_PROJECTS));
+        .lookupPath(ViewIndexDefinition.FIELD_PROJECTS))
+        .cacheKey(cacheKey(viewUuid));
     }
     return viewsFilter;
   }
 
+  public static String cacheKey(String viewUuid) {
```

Comment:
Method name is not appropriated. It's the cache for views lookup. I would name it "viewsLookupCacheKey()"